File: Chapter12/C#/HttpService/Teams/Connection/HttpResponse.cs
using System;
using System.Net.Sockets;

namespace HttpService
{
  /// <summary>
  /// Summary description for HttpResponse.
  /// </summary>
  public class HttpResponse
  {
    Socket socket;

    HttpResponseHeader header;
    public HttpResponseHeader Header
    {
      get {return header;}
    }

    byte[] body;
    public byte[] Body
    {
      get {return body;}
    }

    public int SequenceNumber
    {
      get {return header.SequenceNumber;}
    }

    public HttpResponse(Socket theSocket)
    {
      socket = theSocket;
    }

    public void Get()
    {
      NetworkStream networkStream = new NetworkStream(socket);

      // read the response header
      header = new HttpResponseHeader(networkStream);

      // read the response body (if any)
      if (header.ContentLength == 0) 
        return;

      byte[] buffer = new byte[header.ContentLength];
      networkStream.Read(buffer, 0, buffer.Length);
      body = buffer;
    }
  }
}

File: Chapter12/VB/HttpService/Teams/Connection/HttpResponse.vb
Imports System.Net.Sockets

Public Class HttpResponse
  Private _socket As Socket

  Private _header As HttpResponseHeader
  Public ReadOnly Property Header() As HttpResponseHeader
    Get
      Return _header
    End Get
  End Property

  Private _body As Byte()
  Public ReadOnly Property Body() As Byte()
    Get
      Return _body
    End Get
  End Property

  Public ReadOnly Property SequenceNumber() As Integer
    Get
      Return _header.SequenceNumber
    End Get
  End Property

  Public Sub New(ByVal theSocket As Socket)
    _socket = theSocket
  End Sub

  Public Sub [Get]()
    Dim networkStream As New NetworkStream(_socket)

    ' read the response header
    _header = New HttpResponseHeader(networkStream)

    ' read the response body (if any)
    If _header.ContentLength = 0 Then Return

    Dim buffer(_header.ContentLength - 1) As Byte
    networkStream.Read(buffer, 0, buffer.Length)
    _body = buffer
  End Sub
End Class
